### ISSUE TITLE
Notification and "manual" task system

### ISSUE THREAD

**Issue by espenak** [devilry/devilry-django#411]
We need a notification system for Devilry, and it should have an API that allows us to:
- Display a message that is visible until a sysadmin removes it (downtime notification, important info, ...). Adding such messages would be implemented in an app that uses the notification API.
- Display a message that the user must confirm that they have read (or completed if the message is a manual "task" that the user should perform). This should allow us to set a text that appears in a checkbox that must be checked before the user can say that the task is completed.
- Display a message that disappears automatically when a task is completed. Typical use-case is warning for examiners that they need to correct X students before the exam-registration deadline, or a reminder for course administrators that to mark students that qualify for exams before the exam-registration deadline for their course(s). This kind of message will typically be used by apps, and not added manually.
- Display a message with less important, but useful, information. These messages should have a close/remove button that users can click to remove them. One use-case is to inform about new features in Devilry that users should know about.
## Levels and email notifications

I believe we should have levels, just like logsystems:
- `VERY_IMPORTANT`: Always notified via email.
- `IMPORTANT`: Notified via email, but this can be turned off by the user.
- `INFO`: Not sent via email by default. Should also be possible
## Storage

Messages are stored in the database, and we keep them forever. They will consume very little space since they will be text-only.
## Roles and users

Each message should be tagged with zero or more roles and zero or more users. They are sent to all users with a given role, and all specified users. Note that this is how it should work in the API. UIs are free to implement other methods of selecting the users to send to.
## Appearance

Messages should be shown at the top of the frontpage, and in all the role-dashboards. We should order the messages by their importance and timestamp, and show up to 3 messages with a "show X more messages" link at be bottom of the list. `VERY_IMPORTANT` messages should never be hidden.

Additionally, we should have a message-indicator icon in the top-bar, just like github notifications.
## Nagging

For notification of tasks that the user must complete before the notification disappears, we should support nagging (have you forgotten to complete "Some task" emails). Nagging should increase in frequency until the deadline for a task is reached (E.G.: every month, 2 weeks before, 1 week before, and each day for the last 2 days).

**Comment by torgeirl** [comment]
Related discussions in #995 (Notification settings) and #1011 (Notification center).

**Comment by torgeirl** [comment]
No longer relevant.
